### PR TITLE
Prompt the user to sign in after clicking "See more details"

### PR DESCRIPTION
**Pull request by sampoder**
Previously, for transparent orgs, it'd redirect them to the org's homepage. See #3303, fixes #3303.

### CODE REVIEW

**Review by maxwofford — APPROVED** [review]
Thanks for the fix @sampoder-- merge when you're ready!

**Comment by sampoder** [comment]
let's do it!

**Comment by cjdenio** [comment]
Screenshot 2023-09-15 at 12 17 10 PM

@sampoder this seems to break the button's style + not actually work as intended

**Comment by sampoder** [comment]
Ah sorry about that! Let me try and get it fixed

From Sam

**Comment by sampoder** [comment]
Oh that's interesting it goes to this URL:

```
https://hcb.hackclub.com/users/auth?return_to=https%3A%2F%2Fhcb.hackclub.com%2Fhcb%2FnEH0VZ%3Fredirect_to_sign_in%3Dtrue
```

But then the return_to is lost, on:

```
https://hcb.hackclub.com/users/login_code
```

Maybe this is related to #3439?

File: app/views/hcb_codes/attach_receipt.html.erb [hunk 1]
```diff
@@ -25,7 +25,7 @@
   <hr>
 
   <div class="center">
-    <%= link_to @hcb_code, class: "btn bg-accent center m1" do %>
+    <%= link_to @hcb_code, { redirect_to_sign_in: true }, class: "btn bg-accent center m1" do %>
```
Comment (sampoder) [inline app/views/hcb_codes/attach_receipt.html.erb]: Also curious, what did I do wrong here that causing the styling break? Sorry about that again!

**Pull request merged**